### PR TITLE
WIP > feat: twap form styling WIP

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Styling TWAP container form WIP
- See comments for progress and final update

Import Todo's:
- The limit price in 'Price protection' is hard-coded. Need assistance to add the dynamic calculated limit price.
- We need to add the 'Receive (incl. fee)' element we use for SWAP. Need assistance with this.
Screenshot 2023-07-05 at 15 14 47

### CODE REVIEW

**Comment by fairlighteth** [comment]
Update:
- Added styling
- Limit price is hardcoded.

Screenshot 2023-07-05 at 13 29 13

**Comment by fairlighteth** [comment]
Update: Re-order elements

Screenshot 2023-07-05 at 14 50 24

**Comment by fairlighteth** [comment]
Update
- mobile responsive style fixes

Screenshot 2023-07-05 at 15 12 25
Screenshot 2023-07-05 at 15 12 16

**Comment by alfetopito** [comment]
@fairlighteth I'll take care of the hard coded value

**Comment by fairlighteth** [comment]
Follow up draft PR as a WIP on further styling: https://github.com/cowprotocol/cowswap/pull/2796

**Comment by alfetopito** [comment]
@fairlighteth done on https://github.com/cowprotocol/cowswap/pull/2797

I also did a bit of refactor to make it more generic

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @fairlighteth , great job!

Some issues that I faced in the PR: 
1. I get a crash when I open the app in iOS device (iPhone 12 mini), navigate to the Advanced orders page and connect to MM using WCv1 
image
2. In iOS (real device) in the light mode color of an amount in the To field is different: 
image
3. When I enter a decimal value into the Price protection field, the value is cropped
decimal value
4. Price protection field does not look good in a mobile view
price mobile view
5. It would be great to add upwards and downwards arrows into the Parts field

Thanks!

File: src/modules/trade/pure/TradeNumberInput/index.tsx [hunk 1]
```diff
@@ -12,10 +12,20 @@ export interface TradeNumberInputProps extends TradeWidgetFieldProps {
...
 export function TradeNumberInput(props: TradeNumberInputProps) {
-  const { value, suffix, onUserInput, placeholder = '0', decimalsPlaces = 0, min, max = 100_000 } = props
+  const {
+    value,
+    suffix,
+    onUserInput,
+    placeholder = '0',
+    decimalsPlaces = 0,
+    min,
+    max = 100_000,
```
Comment (anxolin) [inline src/modules/trade/pure/TradeNumberInput/index.tsx:26]: why max is 100K?

Comment (alfetopito) [reply src/modules/trade/pure/TradeNumberInput/index.tsx:26]: https://github.com/cowprotocol/cowswap/pull/2685#issuecomment-1621810939

**Review by elena-zh — APPROVED** [review]
Hey @fairlighteth , great job!
Approving the PR in order not to block the release. 
However, still there is 1 nitpick related to the tooltip icon in a mobile view 
tooltip is shifted

LMK please if I need to create a separate issue for this. 
Thanks

File: src/modules/trade/pure/TradeNumberInput/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/modules/trade/pure/TradeNumberInput/index.tsx:26]: a bit arbitrary, but ooook

**Comment by fairlighteth** [comment]
Updated mobile to look like:
Screenshot 2023-07-11 at 11 27 27

**Comment by fairlighteth** [comment]
Also there should be enough margin in between the InlineBanners

**Comment by elena-zh** [comment]
@fairlighteth , thanks! LGTM!

File: src/modules/trade/pure/TradeNumberInput/index.tsx (hunk 1 above)
Comment (alfetopito) [reply src/modules/trade/pure/TradeNumberInput/index.tsx:26]: 100% arbitrary yes!

Open to suggestions

Comment (alfetopito) [reply src/modules/trade/pure/TradeNumberInput/index.tsx:26]: Is there any reason why we would ever want to have an order with more than 100k parts?

**Pull request merged**